test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,296 @@
+#include <r_util.h>
+#include <r_vector.h>
+#include <r_core.h>
+#include <r_cons.h>
+#include <r_util/r_annotated_code.h>
+
+#include "minunit.h"
+
+static RCodeAnnotation makeRCodeAnnotation(int st, int en, RCodeAnnotationType typec,
+	ut64 offset, RSyntaxHighlightType types) {
+	RCodeAnnotation annotation = {};
+	annotation.start = st;
+	annotation.end = en;
+	annotation.type = typec;
+	if (annotation.type == R_CODE_ANNOTATION_TYPE_OFFSET) {
+		annotation.offset.offset = offset;
+	}
+	if (annotation.type == R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT) {
+		annotation.syntax_highlight.type = types;
+	}
+	return annotation;
+}
+
+static RVector getSomeAnnotationsForAdd() {
```

Comment:
```suggestion
static RVector get_some_code_annotation_for_add(){
```

I also don't think we should return `RVector` as a struct here. It should be `RVector *` with the vector being allocated entirely on the heap, i.e. with `r_vector_new()`.